### DISCUSSION TITLE
nigthly build

### DISCUSSION THREAD

**Discussion by dutrieux** [fomantic/Fomantic-UI#1738] [General] [answered]
Hi,

I have a problem when I try to build the nigthly build :

![image](https://user-images.githubusercontent.com/1622751/97815467-852a2e80-1c8e-11eb-92c1-5a50f1faec1e.png)

and on the result of dist\semantic.css the @import url(https://fonts.googleapis.com/css2?family?...) is missing

![image](https://user-images.githubusercontent.com/1622751/97815488-b60a6380-1c8e-11eb-8332-cf3a40532c48.png)

**Comment by lubber-de** [comment]
How does your semantic.json look like? Could you try to use node 12 or lower? (the github actions build up to node 12)

**Reply by dutrieux** [reply]
I try with node v12.19.0 : same problem.

I don't modify semantic.json : [semantic.zip](https://github.com/fomantic/Fomantic-UI/files/5471868/semantic.zip)

**Reply by lubber-de** [reply]
From where (directory location) do you call `npx gulp build`? I always build from the fomantic root folder (and cannot reproduce unfortunately).

**Reply by dutrieux** [reply]
I try to an other computer (same problem), below from scratch how I try to build the nightly build :

![image](https://user-images.githubusercontent.com/1622751/97852629-454b6180-1cf7-11eb-940c-6cea6203ab05.png)

**Comment by lubber-de** [comment]
Seems you are running `npx gulp build` from `d:\temp\sui\semantic`.
Try from `d:\temp\sui\node_modules\fomantic-ui`

Which gulp version are you using? check by `npx gulp -v`

**Comment by dutrieux** [comment]
![image](https://user-images.githubusercontent.com/1622751/97853103-f18d4800-1cf7-11eb-81eb-40b0f00cddf4.png)

I try to build from `d:\temp\sui\node_modules\fomantic-ui` same problem !

**Comment by ko2in** [comment]
I have the same issue (not nightly, but the standard FUI package) after I upgraded my Ubuntu from 18.04 to 20.04 in last month and I still haven't time to look into this. I think there is some incompatible dependencies that used in FUI and the current version of npm.

**Reply by lubber-de** [reply]
I still cannot reproduce unfortunately using Windows and WSL Ubuntu 20.04
![image](https://user-images.githubusercontent.com/18379884/97878351-d1bd4a80-1d1e-11eb-9910-0d20d159ed8e.png)
![image](https://user-images.githubusercontent.com/18379884/97878376-da158580-1d1e-11eb-9d00-7c03ae8087a2.png)

**Reply by ko2in** [reply]
It only happens under the git repo directory. It doesn't happen when installing with `npm install fomantic-ui` and the gulp build under semantic directory.

![fui-terminal](https://user-images.githubusercontent.com/930315/97888323-68a6f880-1d59-11eb-95cb-71fe85e32985.png)

Here's the environments on my Ubuntu 20.04.

![terminal](https://user-images.githubusercontent.com/930315/97888604-b58acf00-1d59-11eb-8e31-afb807ca3ea7.png)

I've the latest versions than yours. Maybe, you should check with these versions. As far as I remembered, I got this issue after upgrading my Ubuntu version. I've been running `gulp build` under my local git repo all the time and working properly.

**Comment by dutrieux** [comment, marked as answer]
I try with virtualbox and Ubuntu 20.04.1 LTS and I have the same problem : 

![image](https://user-images.githubusercontent.com/1622751/97918920-309eb600-1d57-11eb-8862-bb730960f127.png)

Is there a parameter to see more info (log...)

**Reply by lubber-de** [reply]
@ColinFrick Can you help? We have to think about replacing gulp-concat-css anyway because of #1646

**Reply by ColinFrick** [reply]
Can't reproduce the problem on Mac or LinuxMint. I'll look into VirtualBox

**Reply by ColinFrick** [reply]
@lubber-de was able to reproduce and find the problem. The commit d4f9aa3777f6bb7d3b2eadc1e310a836468a43ef added semicolons to the import url, this causes `parse-import` used by `gulp-concat-css` to fail.
We could revert to v1 API or maybe fork concat-css and parse-import, because I don't see any drop in replacement.

**Reply by lubber-de** [reply]
@ColinFrick I quickly tried to replace the semicolon by `%3B`, which seems to work for the google api.
Could you try this in your environment to make sure it solves the issue as well? (i would like to avoid reverting the change)

So it results in

```
https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,400%3B0,700%3B1,400%3B1,700&subset=latin&display=swap
```

**Reply by lubber-de** [reply]
@dutrieux @ColinFrick @ko2in   I prepared PR #1743. Could you please try this locally where you got the issue? The github actions builds it just fine 🙂

**Reply by dutrieux** [reply]
> @dutrieux @ColinFrick @ko2in I prepared PR #1743. Could you please try this locally where you got the issue? The github actions builds it just fine 🙂

I update locally your PR #1743 and now it's ok, no error message during build and semantic.css have now `@import url('https://fonts.googleapis.com/css2?...)`

Thx everybody for your help

**Reply by ko2in** [reply]
I was about to report what I discovered about this issue. I got only this when there is `site` component in build tasks. If I excluded the `site` component, the error doesn't appear. Here is my `semantic.json`.
```
{
  "base": "",
  "paths": {
    "source": {
      "config": "src/theme.config",
      "definitions": "src/definitions/",
      "site": "src/site/",
      "themes": "src/themes/"
    },
    "output": {
      "packaged": "dist/",
      "uncompressed": "dist/components/",
      "compressed": "dist/components/",
      "themes": "dist/themes/"
    },
    "clean": "dist/"
  },
  "permission": false,
  "autoInstall": false,
  "rtl": false,
  "version": "2.8.7",
	"components": [
		"accordion", "ad", "api", 
		"breadcrumb", "button", 
		"calendar", "card", "checkbox", "comment", "container", 
		"dimmer", "divider", "dropdown", 
		"embed", "emoji", 
		"feed", "flag", "form", 
		"grid", 
		"header", 
		"icon", "image", "input", "item", 
		"label", "list", "loader", 
		"menu", "message", "modal", 
		"nag", 
		"placeholder", "popup", "progress", 
		"rail", "rating", "reset", "reveal", 
    "search", "segment", "shape", "sidebar", "slider", "state", "statistic", "step", "sticky", 
    "tab", "table", "text", "toast", "transition", 
		"visibility"
	]
}
```
I'm sure the PR would definitely fix it since it's inside `site.variables` :wink: . I'll test the PR and let you know.

**Reply by ko2in** [reply]
I confirm that the PR works and the error no longer appearing.

**Reply by lubber-de** [reply]
Merged 😄 Tomorrow, the nightly branch has got the fix implemented
